lib/filterlist-utils.ts

```diff
@@ -0,0 +1,48 @@
+import { FiltersEngine } from '../node_modules/@cliqz/adblocker/src/index';
+import { extractFeaturesFromDOM } from '../node_modules/@cliqz/adblocker-content/src/index';
+import { parse as tldtsParse } from 'tldts-experimental';
+import { serializedEngine } from './filterlist-engine';
+
+export function deserializeFilterList() {
+  return FiltersEngine.deserialize(serializedEngine)
+}
+
+export function parseFilterList(rawFilterlist: string) {
+  const engine = FiltersEngine.parse(rawFilterlist, {
+    enableMutationObserver: false, // we don't monitor DOM changes at the moment
+    loadNetworkFilters: false,
+    enableHtmlFiltering: false,
+    loadCSPFilters: false,
+  });
+  return engine;
+}
+
+export function getCosmeticStylesheet(engine: FiltersEngine): string {
+  const parsed = tldtsParse(location.href);
+  const hostname = parsed.hostname || '';
+  const domain = parsed.domain || '';
+
+  const cosmetics = engine.getCosmeticsFilters({
+    url: location.href,
+    hostname,
+    domain,
+
+    // this extracts current ids, classes and attributes (depends on the current DOM state)
+    ...extractFeaturesFromDOM([document.documentElement]),
+
+    getBaseRules: true,
+    getInjectionRules: false, // we don't inject scripts atm
+    getExtendedRules: true,
+    getRulesFromDOM: true,
+    getRulesFromHostname: true,
+  });
+  return cosmetics.styles;
+}
+
+export function getFilterlistSelectors(styles: string): string {
+  if (styles) {
+    const selectorsOnly = styles.replace(/\s*{[^\\}]*}\s*/g, ',').replace(/,$/, '');
```

Comment:
Could you explain this part?

Reply:
The idea is to strip out all `{...}` parts, so we have just the selectors without the actual CSS rules. Those selectors are then passed to a utility function that determines if there are any "visible" elements matching those.

Reply:
I wonder if it would be better to use the already parsed filters, to pull out the selector part. (Assuming the filter parsing library you used let's you do that.)

Reply:
Could we write a test for this to ensure that it's robust to changes in the output of the external adblocker library? i.e. a simple unit-test that creates a filter engine with dummy filters, gets the selectors and runs this function, emitting an expected list of selectors.